### PR TITLE
feat: zaps

### PR DESCRIPTION
**Pull request by SamSamskies**

### CODE REVIEW

**Comment by SamSamskies** [comment]
@michaelhall923 when you get a chance, can you check out why the preview deploy failed? i don't have access to the project on vercel. this is still a wip, but just wanted to get a preview of it up so people can play with it as i finish the rest.

File: ndk/utils.tsx [hunk 1]
```diff
@@ -240,3 +252,192 @@ export const fetchEvents = async (
...
+  if (isAnonymous) {
+    return finishEvent(event, generatePrivateKey());
+  }
+
+  const zapRequestEvent = new NDKEvent(ndk, event as NostrEvent);
+
+  await zapRequestEvent.sign();
+
+  return await zapRequestEvent.toNostrEvent();
+};
```
Comment (michaelhall923) [inline ndk/utils.tsx:367]: Rename zapRequestEvent to just event

**Review by michaelhall923 — DISMISSED** [review]

File: theme/Theme.tsx [hunk 2]
```diff
@@ -113,13 +113,17 @@ const stemstrTheme: MantineThemeOverride = {
     TextInput: {
       defaultProps: {
         radius: "md",
+        size: "lg",
       },
```
Comment (michaelhall923) [inline theme/Theme.tsx:117]: Something here messed up the input styles on the signup and edit profile pages

File: ndk/utils.tsx (hunk 1 above)
Comment (SamSamskies) [reply ndk/utils.tsx:367]: That's the name of one of the params though.

Comment (SamSamskies) [reply ndk/utils.tsx:367]: I agree the name here doesn't make sense though 😂. Let me think about it.

Comment (SamSamskies) [reply ndk/utils.tsx:367]: How about this?

```ts
const signEvent = async ({ eventTemplate, ndk, isAnonymous }: SignEventParams) => {
  if (isAnonymous) {
    return finishEvent(eventTemplate, generatePrivateKey());
  }

  const event = new NDKEvent(ndk, eventTemplate as NostrEvent);

  await event.sign();

  return await event.toNostrEvent();
};
```

File: theme/Theme.tsx [hunk 3]
```diff
@@ -152,13 +156,21 @@ const stemstrTheme: MantineThemeOverride = {
...
             backgroundColor:
-              params.variant === "default" ? theme.colors.gray[4] : undefined,
+              params.variant === "filled" ? theme.colors.gray[4] : undefined,
+            color:
+              params.variant === "filled"
+                ? theme.fn.rgba(theme.colors.gray[1], 0.3)
+                : undefined,
           },
         },
         label: {
```
[the remark this answers is not in the document]
Comment (SamSamskies) [reply theme/Theme.tsx:176]: does this still need investigation? i noticed you resolved the issue.

[the remark this answers is not in the document]
Comment (michaelhall923) [reply theme/Theme.tsx:176]: yeh i just messed up the code reference so i remade it

**Comment by SamSamskies** [comment]
I just noticed the custom input screen is messed up with long names. Gotta truncate it. Fixing now.

**Comment by michaelhall923** [comment]
Oh cool, didn't know about the truncate prop. could probably use that in note headers too.

**Comment by michaelhall923** [comment]
all looks good to me

**Pull request merged**